packages/typespec-test/test/batch_modular/generated/typespec-ts/src/api/pagingHelpers.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  getPagedAsyncIterator,
+  PagedAsyncIterableIterator as CorePagedAsyncIterableIterator,
+  PagedResult,
+  PageSettings as CorePageSettings,
+} from "@azure/core-paging";
+import {
+  Client,
+  createRestError,
+  PathUncheckedResponse,
+} from "@azure-rest/core-client";
+import {
+  PageSettings,
+  PagedAsyncIterableIterator,
+} from "../models/pagingTypes.js";
+
+export function buildPagedAsyncIterator<
+  TElement,
+  TResponse extends PathUncheckedResponse = PathUncheckedResponse
+>(
+  client: Client,
+  getInitialResponse: () => PromiseLike<TResponse>,
+  processResponseBody: (result: TResponse) => Promise<unknown>
+): PagedAsyncIterableIterator<TElement> {
+  let firstRun = true;
+  let itemName: string, nextLinkName: string | undefined;
+  const firstPageLinkPlaceholder = "";
+  const pagedResult: PagedResult<TElement[]> = {
+    firstPageLink: firstPageLinkPlaceholder,
+    getPage: async (pageLink: string) => {
+      const result =
+        firstRun && pageLink === firstPageLinkPlaceholder
+          ? await getInitialResponse()
+          : await client.pathUnchecked(pageLink).get();
+      if (firstRun) {
+        const pageInfo = getPaginationProperties(result);
+        itemName = pageInfo.itemName;
+        nextLinkName = pageInfo.nextLinkName;
+      }
+      firstRun = false;
+      checkPagingRequest(result);
+      const results = await processResponseBody(result as TResponse);
+      const nextLink = getNextLink(results, nextLinkName);
+      const values = getElements<TElement>(results, itemName);
+      return {
+        page: values,
+        nextPageLink: nextLink,
+      };
+    },
+    byPage: (settings?: PageSettings) => {
+      const { continuationToken } = settings ?? {};
+      return getPageAsyncIterator(pagedResult, {
+        pageLink: continuationToken,
+      }) as any;
```

Comment:
why is this cast needed?

Reply:
I duplicated the references in core so the castcading is no need now.